pkg/registry/apis/provisioning/extras/register.go

```diff
@@ -35,3 +41,24 @@ func ProvideProvisioningOSSRepositoryExtras(
 		),
 	}
 }
+
+func ProvideExtraWorkers(
+	cfg *setting.Cfg,
+	renderer rendering.Service,
+	blobstore resource.ResourceClient,
+	configProvider apiserver.RestConfigProvider,
+) []jobs.Worker {
+	urlProvider := func(_ string) string {
+		return cfg.AppURL
+	}
+
+	clients := resources.NewClientFactory(configProvider)
```

Comment:
Let's have a ProvidePullRequestWorker in `webhooks/pullrequest/register.go`. That will be called by this one. In that way, we keep the logic for the webhook self-contained there. This ExtraWorkers will likely call multiple provider functions in the future or very soon.